subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -45,42 +81,166 @@ void agps_print_enable(bool enable)
 	agps_print_enabled = enable;
 }
 
-static int type_array2str(enum gps_agps_type *types, size_t type_count,
-			       char *type_array, size_t type_array_len)
+static int get_modem_info(struct modem_param_info *const modem_info)
 {
-	size_t len = 0;
+	__ASSERT_NO_MSG(modem_info != NULL);
 
-	for (size_t i = 0; i < type_count; i++) {
-		int err;
-		enum nrf_cloud_agps_type cloud_type = types[i];
+	int err = modem_info_init();
 
-		err = snprintk(&type_array[len], type_array_len,
-				"%d,", (int)cloud_type);
-		if (err < 0) {
-			return err;
-		}
+	if (err) {
+		LOG_ERR("Could not initialize modem info module");
+		return err;
+	}
+
+	err = modem_info_params_init(modem_info);
```

Comment:
```modem_info_init()``` and ```modem_info_params_init()``` is placed in a function that looks to be called "frequently". This looks like a leak since ```at_params_list_init()``` in ```modem_info_init()``` allocates memory for a list that is not freed in between calls to ```get_modem_info()```.

Reply:
interesting, it seems that `modem_info_init()` should be updated to only allow for initialization once?
And/or add `modem_info_uninit()`?
asset tracker also calls `modem_info_init()`

Reply:
updated to only init once.  @bjda do you have any comments on this issue, i believe you have been doing some modem info rework?